### ISSUE TITLE
 [QUESTION] Line breaker after constraints for type lenses

### ISSUE THREAD

**Issue by hellwolf** [haskell/haskell-language-server#3811]
I was referred to here instead from https://github.com/emacs-lsp/lsp-haskell/issues/178.

## Is your enhancement request related to a problem? Please describe.

![image](https://github.com/emacs-lsp/lsp-haskell/assets/186660/3ddd7275-e642-4071-a7d6-333788a94949)

As illustrated above, it seems the codeLens of "ghcide-type-lenses"(?) break the lines right after the constraints section (after "=>"). Is it a known behavior?

## Describe the solution you'd like

Ideally, the line breaking should follow the max line length consistent with the editor settings, or at least configurable too.

## Describe alternatives you've considered

I have yet to dug deeper to see if it's something related to lsp emacs, or HLS itself.

**Comment by michaelpj** [comment]
> Ideally, the line breaking should follow the max line length consistent with the editor settings, or at least configurable too.

I don't believe we have access to this information, sadly. I guess we could make it configurable :thinking: 

I suspect ultimately this is boiling down to some GHC pretty-printing or similar, which I think does have configurable line breaking, so I think in principle we could do this if we knew what length to pick...

**Comment by hellwolf** [comment]
Oh, good that you hinted GHC pretty-printing. I was digging around to understand where it came from...

Also, I do notice that the line break always happen after "=>"; I deduced this from a few other samples.
